src/accountsdb/download.zig

```diff
@@ -227,321 +229,188 @@ pub fn downloadSnapshotsFromGossip(
-                const inc_snapshot_filename = try std.fmt.allocPrint(allocator, "incremental-snapshot-{d}-{d}-{s}.{s}", .{
-                    peer.full_snapshot.slot,
-                    inc_snapshot.slot,
-                    inc_snapshot.hash,
-                    "tar.zst",
+            const inc_archive_file: ?std.fs.File = blk: {
+                // PERF: maybe do this in another thread? while downloading the full snapshot
+                const inc_snapshot = peer.inc_snapshot orelse break :blk null;
+
+                const inc_snapshot_filename = IncrementalSnapshotFileInfo.snapshotArchiveName(.{
+                    .base_slot = peer.full_snapshot.slot,
+                    .slot = inc_snapshot.slot,
+                    .hash = inc_snapshot.hash,
                 });
-                defer allocator.free(inc_snapshot_filename);
-
-                const inc_snapshot_url = try std.fmt.allocPrintZ(allocator, "http://{s}/{s}", .{
-                    rpc_url,
-                    inc_snapshot_filename,
+                const inc_snapshot_url = bFmt("http://{s}/{s}", .{
+                    bStr(&rpc_url), bStr(&inc_snapshot_filename),
                 });
-                defer allocator.free(inc_snapshot_url);
+                const inc_snapshot_uri = std.Uri.parse(inc_snapshot_url.constSlice()) catch {
+                    const url_str = inc_snapshot_url.constSlice();
+                    std.debug.panic("Failed to Upri.parse '{s}'", .{url_str});
+                };
 
-                logger.info().logf("downloading inc_snapshot from: {s}", .{inc_snapshot_url});
-                _ = downloadFile(
+                logger.info().logf(
+                    "downloading inc_snapshot from: {s}",
+                    .{inc_snapshot_url.constSlice()},
+                );
+                break :blk downloadFile(
                     allocator,
                     logger,
-                    inc_snapshot_url,
+                    inc_snapshot_uri,
                     output_dir,
-                    inc_snapshot_filename,
+                    inc_snapshot_filename.constSlice(),
                     // NOTE: no min limit (we already downloaded the full snapshot at a good speed so this should be ok)
                     null,
+                    download_buffer,
                 ) catch |err| {
                     // failure here is ok (for now?)
                     logger.warn().logf("failed to download inc_snapshot: {s}", .{@errorName(err)});
-                    return;
+                    break :blk null;
                 };
-            }
+            };
 
-            // success
             logger.info().logf("snapshot downloaded finished", .{});
-            return;
+            return .{ full_archive_file, inc_archive_file };
         }
     }
 }
 
-const DownloadProgress = struct {
-    file: std.fs.File,
-    min_mb_per_second: ?usize,
+const DOWNLOAD_FILE_BUFFER_SIZE = 1 * BYTE_PER_MIB + 4096;
```

Comment:
can we keep all global consts at the top of the file

Reply:
yeah nbd - i would say its more related to the actual buffer initialization in `downloadSnapshotsFromGossip` (not downloadFile) -- which caused my confusion and then i search at the top of the file where i expect all constants to be and its not there -- would rather keep it consistent and all at the top

Reply:
Well, the buffer initialized in that function is initialized with the constraints defined by `downloadFile` - the latter defines the kind of buffer it wants.

Reply:
Amended as discussed on slack:

26cb7b1d8a8852edd5b6431f80bb8a2c618203a5